File: client/src/layouts/default_layout/menu/menu-items.js
import ManageAccountsIcon from "@mui/icons-material/ManageAccounts";
import Diversity1Icon from "@mui/icons-material/Diversity1";
import SettingsIcon from "@mui/icons-material/Settings";
import AssignmentIcon from "@mui/icons-material/Assignment";
import AssignmentIndIcon from "@mui/icons-material/AssignmentInd";
import FilePresentIcon from "@mui/icons-material/FilePresent";
import AttachMoneyIcon from "@mui/icons-material/AttachMoney";
import TaskAltIcon from "@mui/icons-material/TaskAlt";
import DescriptionIcon from "@mui/icons-material/Description";
import WorkIcon from "@mui/icons-material/Work";
import PunchClockIcon from "@mui/icons-material/PunchClock";
import GavelIcon from "@mui/icons-material/Gavel"; // thêm cho Contracts
import AccountTreeIcon from '@mui/icons-material/AccountTree';


export const menuItems = [
  {
    title: "Management",
    items: [
      {
        label: "Finance",
        icon: AttachMoneyIcon,
        href: "/management/finance",
        roles: ["ADMIN", "MANAGER", "ACCOUNTANT"],
      },
      {

        label: "Accounts",
        icon: ManageAccountsIcon,
        href: "/management/accounts",
        roles: ["ADMIN"],
      },
      {
        label: "Settings",
        icon: SettingsIcon,
        href: "/management/settings",
        roles: ["ADMIN"],
      },
    ],
  },
  {
    title: "Employees",
    items: [
      {
        label: "Employees List",
        icon: AssignmentIndIcon,
        href: "/employees",
        roles: ["ADMIN", "MANAGER", "HR"],

      },
      {
        label: "Contracts",
        icon: FilePresentIcon,
        href: "/contracts",

        roles: ["ADMIN", "MANAGER"],
      },
      {
        label: "Attendance",
        icon: PunchClockIcon,
        href: "/attendance/list",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE", "HR"],

      },
    ],
  },
  {
    title: "Accountant",
    items: [
      {
        label: "Fund",
        icon: AssignmentIndIcon,
        href: "/finance/fund",
        roles: ["ADMIN", "MANAGER", "ACCOUNTANT"],
        roles: ["ADMIN", "MANAGER", "HOD", "ACCOUNTANT"],
      },
      {
        label: "Salary",
        icon: FilePresentIcon,
        href: "/finance/salary",
        roles: ["ADMIN", "MANAGER", "ACCOUNTANT"],
      },
      {
        label: "Cash Advance",
        icon: FilePresentIcon,
        href: "/payment-request",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE", "HR", "ACCOUNTANT", "CHIEFACCOUNTANT"],
      },
      {
        label: "Payment",
        icon: FilePresentIcon,
        href: "/payment",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE", "HR", "ACCOUNTANT", "CHIEFACCOUNTANT"],

      },
    ],
  },
  {
    title: "Utilities",
    items: [
      {
        label: "Departments",
        icon: Diversity1Icon,
        href: "/departments",
        roles: ["ADMIN", "MANAGER", "HOD", "EMPLOYEE"],
      },
      {
        // Mở cho tất cả để HOD/EMPLOYEE đi theo đường Projects
        label: "Projects",
        icon: AccountTreeIcon,
        href: "/management/projects",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE"],
      },
      {
        label: "Dispatches",
        label: "Dispatches", // = Documents
        icon: AssignmentIcon,
        href: "/management/documents",
        roles: ["ADMIN", "MANAGER", "PM", "ACCOUNTANT"],

      },
      // ⛔ ĐÃ BỎ "Tasks" KHỎI MENU NHƯ YÊU CẦU
      {
        label: "Leave Request",
        icon: DescriptionIcon,
        href: "/leave-request",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE", "HR", "ACCOUNTANT", "SECRETARY", "CHIEFACCOUNTANT"],
      },
      {
        label: "Contracts",
        icon: GavelIcon,
        href: "/contracts",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE", "ACCOUNTANT", "HR", "SECRETARY", "CHIEFACCOUNTANT"],

      },
    ],
  },
];

File: mobile/lib/constants/menu_items.dart
import 'package:flutter/material.dart';

class MenuItemData {
  final String label;
  final IconData icon;
  final String route;
  final List<String> roles;

  const MenuItemData({
    required this.label,
    required this.icon,
    required this.route,
    required this.roles,
  });
}

class MenuSection {
  final String title;
  final List<MenuItemData> items;

  const MenuSection({required this.title, required this.items});
}

const List<MenuSection> menuItems = [
  MenuSection(
    title: "Management",
    items: [
      MenuItemData(
        label: "Finance",
        icon: Icons.attach_money,
        route: "/management/finance",
        roles: ["ADMIN", "MANAGER", "ACCOUNTANT"],
      ),
    ],
  ),
  MenuSection(
    title: "Finance",
    items: [
      MenuItemData(
        label: "Funds",
        icon: Icons.savings,
        route: "/accountant/funds",
        roles: [
          "ADMIN",
          "MANAGER",
          "ACCOUNTANT",
          "CHIEFACCOUNTANT",
          "CHIEF_ACCOUNTANT",
        ],
      ),
      MenuItemData(
        label: "Transactions",
        icon: Icons.compare_arrows,
        route: "/accountant/transactions",
        roles: [
          "ADMIN",
          "MANAGER",
          "ACCOUNTANT",
          "CHIEFACCOUNTANT",
          "CHIEFACCOUNTANT",
        ],
      ),
      MenuItemData(
        label: "Cash Advance",
        icon: Icons.payments,
        route: "/accountant/cash-advance",
        roles: [
          "ADMIN",
          "MANAGER",
          "ACCOUNTANT",
          "CHIEFACCOUNTANT",
          "CHIEF_ACCOUNTANT",
          "EMPLOYEE",
        ],
      ),
      MenuItemData(
        label: "Bank & Top-up",
        icon: Icons.account_balance_wallet,
        route: "/accountant/bank-topup",
        roles: [
          "ADMIN",
          "MANAGER",
          "ACCOUNTANT",
          "CHIEFACCOUNTANT",
          "CHIEF_ACCOUNTANT",
        ],
      ),
      
      MenuItemData(
      label: "Salary Summary",
      icon: Icons.receipt_long,
      route: "/accountant/salaries/summary",
      roles: [
        "ADMIN",
        "MANAGER",
        "ACCOUNTANT",
        "CHIEFACCOUNTANT",
        "CHIEF_ACCOUNTANT",
      ],
    ),
    ],
  ),
  MenuSection(
    title: "Utilities",
    items: [
      MenuItemData(
        label: "Projects",
        icon: Icons.account_tree,
        route: "/utilities/projects",
        roles: ["ADMIN", "MANAGER", "PM", "HOD", "EMPLOYEE","SECRETARY"],
      ),
      MenuItemData(
        label: "Dispatches",
        icon: Icons.assignment,
        route: "/management/documents",
        roles: ["ADMIN", "MANAGER", "PM","SECRETARY"],
      ),

      // Tasks chỉ dành cho HOD & EMPLOYEE
      MenuItemData(
        label: "Tasks",
        icon: Icons.task_alt,
        route: "/utilities/tasks",
        roles: ["HOD", "EMPLOYEE","SECRETARY"],
      ),

      MenuItemData(
        label: "Leave Request",
        icon: Icons.description,
        route: "/utilities/leave-request",
        roles: [
          "ADMIN",
          "MANAGER",
          "PM",
          "HOD",
          "EMPLOYEE",
          "HR",
          "ACCOUNTANT",
          "SECRETARY",
          "CHIEFACCOUNTANT",
        ],
      ),
      MenuItemData(
        label: "Missing Check-out",
        icon: Icons.report_gmailerrorred,
        route: "/attendance/missing-checkout",
        roles: ["HR", "ADMIN"],
      ),
      MenuItemData(
        label: "AttenDance",
        icon: Icons.access_time,
        route: "/attendance/list",
        roles: [
          "ADMIN",
          "MANAGER",
          "PM",
          "HOD",
          "EMPLOYEE",
          "ACCOUNTANT",
          "HR",
          "CHIEFACCOUNTANT"
          "SECRETARY",
        ],
      ),
    ],
  ),
];
